enhancements/sdk-fbc-run-bundle.md

```diff
@@ -0,0 +1,279 @@
+
+#### Prototype Links
+
+- [PR](<https://github.com/operator-framework/operator-sdk/pull/5671>) that implements FBC support in run bundle.
+- [run bundle prototype branch](<https://github.com/rashmigottipati/operator-sdk/tree/run-bundle-prototype>).
+- Initial [POC](<https://github.com/rashmigottipati/run-bundle-POC>) to create a declarative config using Operator Registry APIs.
+
+### Risks and Mitigations
+
+Should we be pinning the upstream opm image `quay.io/operator-framework/opm:` or always use latest.
+    - Consider pinning this image to downstream builds.
+
+The following are some of the risks to using this image:
+
+  1. if image doesn't have bash, we can't modify the entrypoint.
+  1. if opm is building index images without bash, then support for FBC based images will only exist.
+
+## Design Details
+
+### Test Plan
+
+- Manual testing with different scenarios.
+  
+  - Installing an operator bundle when a user provides the operator bundle image they intend to install. This is the default case.
+  - Installing an operator bundle when a user provides an index that is a File-Based Catalog (FBC) image.
+  - Installing an operator bundle when a user provides an index that is a SQLite based image.
+  - Installing an operator bundle when the FBC index provided by the user already has the operator bundle in it. This should error out.
+  - Installing an operator bundle when the SQLite index provided by the user already has the operator bundle in it. This should cause the registry pod to fail with the error `Bundle already exists, Bundle already added that provides package and csv`.
+
+- Unit tests for the new FBC changes added to the implementation (with coverage requirements).
+
+- Possibly e2e tests as well (e2e tests don’t exist for the current run bundle command).
+
+### Graduation Criteria
+
+The `run bundle` command with fbc support will come in as non-alpha and be ready to use once merged.
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+## Drawbacks
+
+N/A
+
+## Alternatives
+
+There is a potential alternative to handling SQLite based images. Instead of taking the old route of using `opm registry add` and `opm registry serve` commands during the creation of ephemeral registry pod, we could render the SQLite based image using Operator Registry APIs or migrate the SQLite image to the new FBC format and serve the contents through the registry pod, which will trigger the installation of an operator bundle. Essentially, treat both images to be the same by rendering and serving the contents. Although this would require dealing with additional compilation to render the SQLite images to FBC. This is a potential alternative to when the `opm commands` such as `opm registry add` and `opm registry serve` are deprecated and support for those will be removed.
```

Comment:
Not sure if I follow up on the latest statement. See that when the deprecated options are removed that means we no longer support SQL index in OPM so why we would need to migrate the old format (SQL) to the new one (fbc)?

Reply:
Added pros and cons in [commit](https://github.com/operator-framework/enhancements/pull/112/commits/3ec0c649c9004bc3b6b9126a4ed37926a02a6cf3)